components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,215 @@
+        ? "WHERE " + column + ">" + lastResult
+        : ""} ORDER BY ${column} DESC`;
+      return this.executeQuery(query);
+    },
+    /**
+     * Gets a single row from a table based on a lookup column & value
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Column to filter by value
+     * @param {string} value - A column value to search for
+     * @returns A single database row
+     */
+    async findRowByValue(table, column, value) {
+      const rows = await this.executeQuery(`SELECT * FROM ${table} WHERE ${column} = '${value}'`);
+      return rows[0];
+    },
+    /**
+     * Gets the number of rows in a table
+     * @param {string} table - Name of database table to query
+     * @returns Count of how many rows exist in the table
+     */
+    async getRowCount(table) {
+      const response = await this.executeQuery(`SELECT COUNT(*) FROM ${table}`);
+      return response[0].count;
+    },
+    /**
+     * Deletes a row or rows in a table based on a lookup column & value
+     * @param {string} table - Name of database table to delete from
+     * @param {string} column - Column used to find the row(s) to delete
+     * @param {string} value - A column value. Used to find the row(s) to delete
+     */
+    async deleteRows(table, column, value) {
+      await this.executeQuery(`DELETE FROM ${table} WHERE ${column} = '${value}'`);
+    },
+    /**
+     * Inserts a row in a table
+     * @param {string} table - Name of database table to insert row into
+     * @param {array} columns - Array of column names
+     * @param {array} values - Array of values corresponding to the column names provided
+     * @returns The newly created row
+     */
+    async insertRow(table, columns, values) {
+      return this.executeQuery(`INSERT INTO ${table} (${columns}) VALUES (${values.map((v) => "'" + v + "'")}) RETURNING *`);
+    },
+    /**
+     * Updates a row in a table
+     * @param {string} table - Name of database table to update
+     * @param {string} lookupColumn - Column used to find row to update
+     * @param {string} lookupValue - A column value. Used to find row to update
+     * @param {object} rowValues - An object with keys representing column names
+     *  and values representing new column values
+     * @returns The newly updated row
+     */
+    async updateRow(table, lookupColumn, lookupValue, rowValues) {
+      const primaryKey = await this.getPrimaryKey(table);
+      let query = `UPDATE ${table} SET`;
+      Object.entries(rowValues).forEach(([
+        key,
+        val,
+      ]) => {
+        query += ` ${key} = '${val}',`;
+      });
+      // trim last comma
+      query = query.slice(0, -1);
+      query += ` WHERE ${primaryKey} =(SELECT ${primaryKey} FROM ${table} WHERE ${lookupColumn} = '${lookupValue}' ORDER BY ${primaryKey} LIMIT 1) RETURNING *`;
+      const response = await this.executeQuery(query);
+      return response[0];
+    },
```

Comment:
If you change `executeQuery` the way I suggested previously I would refactor this one to

```suggestion
    async updateRow(table, lookupColumn, lookupValue, rowValues) {
      const primaryKey = await this.getPrimaryKey(table);
      const columnsPlaceholders = this.getColumnsPlaceholders({
        rowValues,
        fromIndex: 2,
      });
      const response = await this.executeQuery({
        text: format(`
          UPDATE %I SET ${columnsPlaceholders}
            WHERE %I = (
              SELECT %I FROM %I
              WHERE %I = $1
              ORDER BY %I LIMIT 1
            ) RETURNING *
        `, table, primaryKey, primaryKey, table, lookupColumn, primaryKey),
        values: [
          lookupValue,
          ...Object.values(rowValues),
        ],
      });
      return response[0];
    },
```

with this additional function which will give you the necessary placeholder values for the query
```javascript
    getColumnsPlaceholders({
      rowValues = {}, fromIndex = 1,
    }) {
      return Object.keys(rowValues)
        .map((key, index) => `${key} = $${index + fromIndex}`);
    },
```